### PR TITLE
Move span resource name to method

### PR DESCRIPTION
**Pull request by giancarlocosta**
Name span resource via method, similar to the service_name method.

### CODE REVIEW

**Comment by marcotc** [comment]
Looks good @giancarlocosta!
CI seems to be failing because of linting errors. Would you mind running `rubocop` on your branch and addressing any issues?

You'll likely get a green build after that 👍

**Comment by giancarlocosta** [comment]
@marcotc Thanks! Looks good now except for `deploy prelease Gem`. Anything I need to do for that?

**Comment by marcotc** [comment]
@giancarlocosta  Don't worry about `deploy prerelease Gem`, it currently only works for external contributors.

**Comment by marcotc** [comment]
Just one small thing: we want our PRs to be based on the latest unreleased version branch, which currently is `0.28-dev`.
Would you mind rebasing this PR into `0.28-dev` instead of `master`?

File: lib/ddtrace/contrib/faraday/middleware.rb [hunk 1]
```diff
@@ -72,6 +72,10 @@ def service_name(env)
           options[:service_name]
         end
 
+        def resource_name(env)
```
Comment (delner) [inline lib/ddtrace/contrib/faraday/middleware.rb]: @giancarlocosta I think this seems totally fine: what's the motivation for doing so? I imagine you want to patch/override this to customize resource names?

Comment (giancarlocosta) [reply lib/ddtrace/contrib/faraday/middleware.rb]: @delner Yeah that's the goal! 👍

Comment (delner) [reply lib/ddtrace/contrib/faraday/middleware.rb]: @giancarlocosta Okay, ff that's the case, I think that's totally fine, but keep in mind that this component is not considered a part of the public API and therefore might be subject to change in the future without warning. Just want to make sure that risk is understood and acceptable given such a change might break your instrumentation.

**Comment by giancarlocosta** [comment]
@delner @marcotc Thanks guys! Do you have an estimate of when you'll be able to merge this? I looked on the guidelines but didn't see anything about merge times

**Comment by delner** [comment]
@giancarlocosta We'll consider this for our next minor release, which usually happen on Tuesdays. We will sometimes skip a Tuesday if we don't think we have enough substantial changes to ship.

**Pull request merged**